vortex-buffer/src/lib.rs

```diff
@@ -36,6 +39,32 @@ unsafe impl Send for Buffer {}
 unsafe impl Sync for Buffer {}
 
 impl Buffer {
+    /// Create a new buffer of the provided length with all bytes set to `0u8`.
+    /// If len is 0, does not perform any allocations.
+    pub fn from_len_zeroed(len: usize) -> Self {
+        Self::Arrow(ArrowBuffer::from(ArrowMutableBuffer::from_len_zeroed(len)))
+    }
+
+    /// Create a new buffer from a vector.
+    ///
+    /// # Panics
+    /// This method will panic if the vector is empty.
+    #[allow(clippy::as_ptr_cast_mut)]
+    pub fn from_vec<T: Sized + Send + Sync + RefUnwindSafe + 'static>(input: Vec<T>) -> Self {
+        if input.is_empty() {
+            return Self::from_len_zeroed(0);
+        }
+
+        unsafe {
+            ArrowBuffer::from_custom_allocation(
```

Comment:
You don't need this - Buffer::from can take a vec

Reply:
custom allocation is not mutable

Reply:
but only a Vec<u8>

Reply:
only if T is `ArrowNativePType`

Reply:
yeah, you need to put ArrowNativeType bound which is annoying but imho better. Those buffers are immutable which loses out on performance. If we want to get rid of arrow in api we have to have our own buffer type